# *MAPT* expression is mediated by long-range interactions with *cis*-regulatory elements



### To view all analyses click [here](https://aanderson54.github.io/MAPT_cre/)

### Pre-print is available on bioRxiv  [here](https://www.biorxiv.org/content/10.1101/2023.03.07.531520v2)

<br />

## Abstract

![Figrue1](https://github.com/aanderson54/MAPT_cre/blob/main/images/ExperimentalOverview.png)

**Background** Tauopathies are a group of neurodegenerative diseases driven by abnormal aggregates of tau, a microtubule associated protein encoded by the MAPT gene. MAPT expression is absent in neural progenitor cells (NPCs) and increases during differentiation. This temporally dynamic expression pattern suggests that MAPT expression is controlled by transcription factors and cis-regulatory elements specific to differentiated cell types. Given the relevance of MAPT expression to neurodegeneration pathogenesis, identification of such elements is relevant to understanding genetic risk factors.

**Method**s We performed HiC, chromatin conformation capture (Capture-C), single-nucleus multiomics (RNA-seq+ATAC-seq), bulk ATAC-seq, and ChIP-seq for H3K27Ac and CTCF in NPCs and neurons differentiated from human iPSC cultures. We nominated candidate cis-regulatory elements (cCREs) for MAPT in human NPCs, differentiated neurons, and pure cultures of inhibitory and excitatory neurons. We then assayed these cCREs using luciferase assays and CRISPR interference (CRISPRi) experiments to measure their effects on MAPT expression. Finally, we integrated cCRE annotations into an analysis of genetic variation in AD cases and controls.

**Results** Using orthogonal genomics approaches, we nominated 94 cCREs for MAPT, including the identification of cCREs specifically active in differentiated neurons. Eleven regions enhanced reporter gene transcription in luciferase assays. Using CRISPRi, 5 of the 94 regions tested were identified as necessary for MAPT expression as measured by RT-qPCR and RNA-seq. Rare and predicted damaging genetic variation in both nominated and confirmed CREs was depleted in AD cases relative to controls (OR = 0.40, p = 0.004), consistent with the hypothesis that variants that disrupt MAPT enhancer activity, and thereby reduce MAPT expression, may be protective against neurodegenerative disease.

**Conclusions** We identified both proximal and distal regulatory elements for MAPT and confirmed the regulatory function for several regions, including three regions centromeric to MAPT beyond the well-described H1/H2 haplotype inversion breakpoint. This study provides compelling evidence for pursuing detailed knowledge of CREs for genes of interest to permit better understanding of disease risk.
