# *MAPT* expression is mediated by long-range interactions with *cis*-regulatory elements



### Scripts for nomination and CRISPRi validation [here](https://aanderson54.github.io/MAPT_cre/)
Additional scripts for the burden analysis can be found in the `Burden_analysis` folder

### The manuscript is available  [here](https://doi.org/10.1016/j.ajhg.2023.12.015)

<br />

## Abstract

Tauopathies are a group of neurodegenerative diseases definedriven by abnormal aggregates of tau, a microtubule associated protein encoded by the MAPT gene. MAPT expression is near absent in neural progenitor cells (NPCs) and increases during differentiation. This temporally dynamic expression pattern suggests that MAPT expression could beis controlled by transcription factors and cis-regulatory elements specific to differentiated cell types. Given the relevance of MAPT expression to neurodegeneration pathogenesis, identification of such elements is relevant to understanding disease risk and pathogenesis. Here, we performed chromatin conformation assays (HiC & Capture-C), single-nucleus multiomics (RNA-seq+ATAC-seq), bulk ATAC-seq, and ChIP-seq for H3K27Ac and CTCF in NPCs and differentiated neurons to nominate candidate cis-regulatory elements (cCREs). We assayed these cCREs using luciferase assays and CRISPR interference (CRISPRi) experiments to measure their effects on MAPT expression. Finally, we integrated cCRE annotations into an analysis of genetic variation in neurodegeneration cases and controls. We identified both proximal and distal regulatory elements for MAPT and confirmed the regulatory function for several regions, including three regions centromeric to MAPT beyond the H1/H2 haplotype inversion breakpoint. We also found that rare and predicted damaging genetic variation in both nominated and confirmed CREs was nominally depleted in dementia cases relative to controls (OR = 0.40, p = 0.004), consistent with the hypothesis that variants that disrupt MAPT enhancer activity, and thereby reduce MAPT expression, may be protective against neurodegenerative disease. Overall, this study provides compelling evidence for pursuing detailed knowledge of CREs for genes of interest to permit better understanding of disease risk.
